L/libsasl2/build_tarballs.jl

```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libsasl2"
+version = v"2.1.28"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/cyrusimap/cyrus-sasl/releases/download/cyrus-sasl-$(version)/cyrus-sasl-$(version).tar.gz",
+                  "7ccfc6abd01ed67c1a0924b353e526f1b766b21f42d4562ee635a8ebfc5bb38c")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/cyrus-sasl*
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} ac_cv_gssapi_supports_spnego=yes
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("x86_64", "macos"; ),
+    Platform("aarch64", "macos"; ),
+    Platform("x86_64", "freebsd"; )
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libsasl2", :libsasl2),
+    LibraryProduct("libcrammd5", :libcrammd5, "lib/sasl2"),
+    LibraryProduct("libanonymous", :libanonymous, "lib/sasl2"),
+    LibraryProduct("libscram", :libscram, "lib/sasl2"),
+    LibraryProduct("libplain", :libplain, "lib/sasl2"),
+    LibraryProduct("libdigestmd5", :libdigestmd5, "lib/sasl2"),
+    LibraryProduct("libotp", :libopt, "lib/sasl2")
```

Comment:
Uhm, I think these are plugins, not shared libraries.

Reply:
Okay! I'm not 100% sure of the difference, buit I can remove them

Reply:
The difference is subtle, and even more so from a Julia point-of-view since we effectively treat shared libraries as plugins.  Here's an overview about plugins in C: https://eli.thegreenplace.net/2012/08/24/plugins-in-c.  But crucially, plugins typically always have `.so` extension, while we expect `LibraryProduct`s to have platform-specific extensions, so for example on macOS they should have `.dylib` extension, and `.dll` on Windows.